alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxow.hs

```diff
@@ -310,8 +310,17 @@ alonzoStyleWitness = do
          in seq (rnf ans) ans
   null unredeemed ?! UnRedeemableScripts unredeemed
 
-  {-  || { (sp, h) ∈ \fun{scriptsNeeded} utxo tx | h\mapsto s ∈ txscripts txw, s ∈ ScriptPhTwo } || = || fun{txrdmrs} tx ||  -}
-  let rdptrs = Set.fromList [el | (sp, _) <- sphs, SJust el <- [rdptr @era txbody sp]]
+  {-  ∀ (ptr \mapsto _ ) ∈ txrdmrs tx, ptr ∈ { rdptr txb sp | (sp, h) ∈ scriptsNeeded utxo tx,
+    h \mapsto s ∈ txscripts txw, s ∈ Scriptph2}  -}
+  let rdptrs =
+        Set.fromList
+          [ el
+            | (sp, hs) <- sphs,
+              SJust el <- [rdptr @era txbody sp],
+              (hs', script) <- Map.toList (getField @"scriptWits" tx),
+              hs == hs',
+              (not . isNativeScript @era) script
+          ]
       extraRdmrs :: [RdmrPtr]
       extraRdmrs = Map.keys $ Map.withoutKeys (unRedeemers $ txrdmrs $ wits tx) rdptrs
```

Comment:
This is only one of the two inclusions required. It seems like the other one is handled by the previous block using `checkScriptData`, but we removed that function from the spec entirely. So instead, there should be a `missingRdmrs` that's defined the same way as `extraRdmrs`, except that the two sets here are flipped.